kernel/syscall.c

```diff
@@ -2933,6 +2933,30 @@ long myst_syscall_prlimit64(
             old_rlim->rlim_max = MYST_PROCESS_MAX_STACK_SIZE;
         }
     }
+    else if (resource == RLIMIT_NPROC)
+    {
+        if (old_rlim)
+        {
+            old_rlim->rlim_cur =  __myst_kernel_args.max_threads;
+            old_rlim->rlim_max =  __myst_kernel_args.max_threads;
+        }
+    }
+    else if (resource == RLIMIT_AS)
+    {
+        if (old_rlim)
+        {
+            old_rlim->rlim_cur =  -1;
```

Comment:
Is -1 a value with any special meaning here?

Reply:
Good call out, it means infinity and turns out there is a defined constant (RLIM_INFINITY).